File: js/튜플.js
function solution(s) {
    const result = [];
    let subArray = [];
    let temp = '';
    for (let i = 0; i < s.length; i++) {
        const x = s[i];
        if (x === '{') continue;
        if (!isNaN(x)) temp += x;
        if ((x === ',' || x === '}') && temp !== '') {
            subArray.push(parseInt(temp));
            temp = '';
            if (x === '}') {
                result.push(subArray)
                subArray = [];
            }
        }
    }
    
    result.sort((a, b) => a.length - b.length)
    const answer = [];
    for (let arr of result) {
        for (let x of arr) {
            if (!answer.includes(x)) {
                answer.push(x)
            }
        }
    }
    return answer
}

s = "{{4,2,3},{3},{2,3,4,1},{2,3}}"
console.log(solution(s))

File: python/kakao대비/튜플.py
def solution(s):
    arr = sorted([list(map(int, x.split(','))) for x in s[2:-2].split('},{')], key=lambda x: len(x))
    result = []
    for subset in arr:
        for num in subset:
            if num not in result:
                result.append(num)
                break
    return result

print(solution("{{1,2,3},{2,1},{1,2,4,3},{2}}"))